moto/server.py

```diff
@@ -14,9 +14,13 @@
 from werkzeug.serving import run_simple
 
 from moto.backends import BACKENDS
+from moto.compat import load_module
+from moto.core.publisher import default_publisher
 from moto.core.utils import convert_flask_to_httpretty_response
 
+
 HTTP_METHODS = ["GET", "POST", "PUT", "DELETE", "HEAD", "PATCH"]
+ADDITIONAL_URL_BASES = [r"https?://{0}\.(.+)\.xip\.io"]  # see http://xip.io/
```

Comment:
This is a small feature. I'd like to start moto in server mode with support for all backends (instead of single service).
However for that to be usable it will require hacking around with DNS or /etc/hosts.. so all the *.amazon.aws names resolve to localhost.
With this change I can set up the endpoints in the client like `http://rds2.127.0.0.1.xip.io`, `http://ec2.127.0.0.1.xip.io`, etc, etc. And moto server will be able to forward requests to the corresponding backends.